### ISSUE TITLE
Removing an element’s PHP class breaks the CMS editor

### ISSUE THREAD

**Issue by kinglozzer** [silverstripe/silverstripe-elemental#708]
```
TypeError: Cannot read property 'title' of undefined
    at t.value (bundle.js?m=1557373499:1)
    at Ln (vendor.js?m=1560140517:1)
    at Dn (vendor.js?m=1560140517:1)
    at Nn (vendor.js?m=1560140517:1)
    at Jn (vendor.js?m=1560140517:1)
    at Xn (vendor.js?m=1560140517:1)
    at _r (vendor.js?m=1560140517:1)
    at hr (vendor.js?m=1560140517:1)
    at dr (vendor.js?m=1560140517:1)
    at rr (vendor.js?m=1560140517:1)
```

graphql response looks like:

<img width="855" alt="Screenshot 2019-08-01 at 16 02 10" src="https://user-images.githubusercontent.com/1655548/62304376-c58c7280-b475-11e9-88c8-9f23e4787de1.png">

Manually deleting the records (e.g. id 117 from above) fixes it

## Acceptance criteria
- [x] The elemental area still renders
- [x] Blocks of deleted classes are not rendered on the front end
- [x] Blocks of deleted classes are displayed with a warning in the CMS. Users can not edit them, but can delete them.
- [x] We seek feedback from designers and bespoke teams on the best way to communicate the situation to the end user.

## PRs
- https://github.com/silverstripe/silverstripe-elemental/pull/991

**Comment by robbieaverill** [comment]
What should we do here? The CMS says "this page type is obsolete" and I guess changes its class to something more generic on save (whatever is populated first in the class type dropdown). Do we need to do something similar for elements?

We need to handle this error gracefully anyway, maybe a "couldn't render this element: [error message here]" or something, assuming there is an error message.

cc @silverstripeux re: "How do we display elements in the editor that have had their PHP classes removed in code"

**Comment by clarkepaul** [comment]
I like the idea of the error message but being able to extract the raw data so it could be copypasted into another block, and for the frontend we don't show anything.

My thought would be to show a collapsed block with the error message, expand it to show the table data (doesn't need to be pretty at all) with any additional error info if available.

**Comment by kinglozzer** [comment]
Looking at what the graphql endpoint returns, I wonder if the original intention was for the block editor to display a generic “Block” as the block type... so maybe this a regression?

If it helps, the error was coming from this line specifically: https://github.com/dnadesign/silverstripe-elemental/blob/8a2b9db84c7c4b0ef51c216514613476c223bc10/client/src/components/ElementEditor/Element.js#L192

Thankfully this doesn’t cause any errors on the frontend, the affected blocks are just missing.

**Comment by gdelavil** [comment]
Hi @kinglozzer @robbieaverill ! I was wondering if you guys have any updates on this bug? It's still happening in v4.3.1, and it's making the deprecation of a component a bit difficult.

**Comment by robbieaverill** [comment]
I think we need some sort of task that runs on flush which can re-map obsolete elemental classes in the DB to something like ElementContent that always exists. Then we could show in the UI that the old element type is gone, like we do for pages. This would allow us to implement the UI in the way @clarkepaul described above.

In terms of updates, there's been no progress made on this issue. Pull requests would be welcomed if you wanted to give it a crack in the meantime!

**Comment by prawn185** [comment]
Had this issue today as well
**Aim**: Rename a component i.e from SomethingAbigious -> SlideShow
**Repercussions:** Whole page doesn't display elemental Block section (Where you can add blocks and click into them) - No errors
**Tried:**
Dev/build
**Solution:**
Renaming component back to SomethingAbigious removing it and renaming back to SlideShow
I feel as though it needs to check if that element exists any more and not try to show it?

**Comment by lerni** [comment]
I run into it several times. Usually if the whole `ElementalArea` breaks this or wrong `provideBlockSchema()`in one of the used `Elements` is the cause. **The "no error" part makes it a bummer :(**

**Comment by sabina-talipova** [comment]
Solution was manually tested in local environment. Everything works perfect.
Close issue.

**Comment by gdelavil** [comment]
Hi @sabina-talipova . Do you know in which version of Silverstripe this fix would be released?

**Comment by GuySartorelli** [comment]
@gdelavil Released as [4.9.2](https://github.com/silverstripe/silverstripe-elemental/releases/tag/4.9.2)
